crates/cheatcodes/src/inspector.rs

```diff
@@ -1171,6 +1168,10 @@ impl<DB: DatabaseExt> Inspector<DB> for Cheatcodes {
             }
         }
 
+        // allow cheatcodes from the address of the new contract
+        // Compute the address *after* any possible broadcast updates, so it's based on the updated
+        // call inputs
+        let address = self.allow_cheatcodes_on_create(data, call);
```

Comment:
Note that this changes the old, possibly incorrect, behavior. It might not matter, but now wallet deployed accounts are now allowed to executed cheatcodes.